codec_test.go

```diff
@@ -22,17 +22,17 @@ var encoderTests = []struct {
 }
 
 func TestRoundTrip(t *testing.T) {
-	buf := new(bytes.Buffer)
-	enc := NewEncoder(buf, false)
-	dec := NewDecoder(buf)
 	for _, tt := range encoderTests {
+		buf := new(bytes.Buffer)
+		enc := NewEncoder(buf, false)
 		want := tt.event
 		if err := enc.Encode(want); err != nil {
 			t.Fatal(err)
 		}
 		if buf.String() != tt.output {
 			t.Errorf("Expected: %s Got: %s", tt.output, buf.String())
 		}
+		dec := NewDecoder(buf, 0)
```

Comment:
I had to move this because in the new implementation, the decoder starts trying to read from the reader as soon as you create it, and reading from an empty buffer would cause an immediate EOF.